src/main/java/model/FruitTransaction.java

```diff
@@ -0,0 +1,30 @@
+package model;
+
+public class FruitTransaction {
+    private final Operation operation;
+    private final String fruit;
+    private final int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        if (operation == null || fruit == null || fruit.isEmpty()) {
+            throw new IllegalArgumentException("Invalid argument(s) for creating FruitTransaction");
+        }
```

Comment:
let's not add these validations to the constructor of this class. Validate these params outside the class